### PR TITLE
Use thor instead of OptionParser

### PR DESCRIPTION
**Pull request by estahn**
[thor](http://whatisthor.com) is an (arguably) nicer  alternative to OptionParser. Have a look and let me know if you would like to pursue thor rather then OptionParser.

### CODE REVIEW

**Comment by skywinder** [comment]
It looks very neat and self-descriptive, unlike OptionParser. So, I think it's a good idea to go with `thor`! :+1:

**Comment by tya** [comment]
+1 for this change.  This auto version bump tool also uses thor: https://github.com/RiotGamesMinions/thor-scmversion

**Comment by skywinder** [comment]
@tya Ok. I will try to play around this versionIgniting system as well. Thanks you for suggestion!
I don't forget about this PR, @estahn take a great work to improve this project and I so appreciate that! As soon as I complete my currents tasks - I will handle this PR.

File: github_changelog_generator.gemspec [hunk 1]
```diff
@@ -28,4 +28,5 @@ Gem::Specification.new do |spec|
 
   spec.add_runtime_dependency('github_api', ['~> 0.12'])
   spec.add_runtime_dependency('colorize', ['~> 0.7'])
+  spec.add_runtime_dependency('thor', ['~> 0.19'])
 end
```
Comment (martymcfly23) [inline github_changelog_generator.gemspec]: /> <H3> ###

Comment (skywinder) [inline github_changelog_generator.gemspec]: @martymcfly23 what does it mean?

**Comment by skywinder** [comment]
Thanks you, @estahn 
Thor is incredibly powerful feature, but seems it's not time for that merge. 👍

**Pull request closed without merging**